tests/functional/test_endpoints.py

```diff
@@ -171,3 +173,45 @@ def _assert_service_name_matches_endpoint_prefix(session, service_name):
         "Actual service name `%s` does not match expected service name "
         "we computed: `%s`" % (
             actual_service_name, computed_name))
+
+
+class TestEndpointResolution(BaseSessionTest):
+
+    def create_stubbed_client(self, service_name, region_name, **kwargs):
+        client = self.session.create_client(service_name, region_name, **kwargs)
+        http_stubber = ClientHTTPStubber(client)
+        http_stubber.start()
+        http_stubber.add_response()
+        return client, http_stubber
+
+    def test_regionalized_client_endpoint_resolution(self):
+        client, stubber = self.create_stubbed_client('s3', 'us-east-2')
+        client.list_buckets()
+        self.assertEquals(
+            stubber.requests[0].url,
+            'https://s3.us-east-2.amazonaws.com/'
+        )
+
+    def test_regionalized_client_with_unknown_region(self):
+        client, stubber = self.create_stubbed_client('s3', 'not-real')
+        client.list_buckets()
+        # Validate we don't fall back to partition endpoint for
+        # regionalized services.
+        self.assertEquals(
+            stubber.requests[0].url,
+            'https://s3.not-real.amazonaws.com/'
+        )
+
```

Comment:
On a similar note, we should add more tests for `sts` to make sure that unknown regions go to the regional endpoint and not the global one. Specifically, I was thinking we add the test cases in `tests/functional/test_sts.py` and make sure that the `AWS_STS_REGIONAL_ENDPOINTS` configuration always results in using the regional endpoint of an unknown sts region.